query/expression.go

```diff
@@ -230,6 +246,26 @@ func (expr *functionExpression) Evaluate(context EvaluationContext) (value, erro
 		return seriesListValue(series), nil
 	}
 
+	if name == "timeshift" {
+		// A timeshift performs a modification to the evaluation context.
+		// In the future, it may be one of a class of functions which performs a similar modification.
+		// A timeshift has two parameters: its first (which it evaluates), and its second (the time offset).
+		if len(expr.arguments) != 2 {
+			return nil, errors.New(fmt.Sprintf("Function `timeshift` expects 2 parameters but is given %d (%+v)", len(expr.arguments), expr.arguments))
+		}
+		shift, err := expr.arguments[1].Evaluate(context)
+		if err != nil {
+			return nil, err
+		}
+		durationNano, err := toDuration(shift)
+		if err != nil {
+			return nil, err
+		}
+		newContext := context
+		newContext.Timerange = newContext.Timerange.Later(durationNano)
+		return expr.arguments[0].Evaluate(newContext)
```

Comment:
we would want to rewrite the timerange of the response. to be the current timestamp, not the shifted timestamp.